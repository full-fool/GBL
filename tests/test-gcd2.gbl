class mymain extends Main{
int gcd(int a, int b) {
  while (a <= b)
    if (a > b) a = a - b;
    else b = b - a;
  return a;
}

int main()
{
  printlni(gcd(14,21));
  printlni(gcd(8,36));
  printlni(gcd(99,121));
  return 0;
}
}