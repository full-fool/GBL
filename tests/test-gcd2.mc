int gcd(int a, int b) {
  while (a neq b)
    if (a > b) a = a - b;
    else b = b - a;
  return a;
}

int main()
{
  println(gcd(14,21));
  println(gcd(8,36));
  println(gcd(99,121));
  return 0;
}
