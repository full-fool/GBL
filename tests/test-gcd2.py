import sys
class mymain:
    def __init__(self):
        pass

    def main(self):
        print (self.gcd(14, 21))
        print (self.gcd(8, 36))
        print (self.gcd(99, 121))
        return 0
        pass

    def gcd(self,a,b):
        while ((a) <= (b)):
            if ((a) > (b)):
                a = (a) - (b)
            else:
                b = (b) - (a)
        return a
        pass


if __name__ == "__main__":
    _main = mymain()
    _main.main()

