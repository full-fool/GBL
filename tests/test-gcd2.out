14
8
99
